assets/test/socket_test.js

```diff
@@ -632,7 +632,9 @@ describe("onConnClose", () => {
   it("schedules reconnectTimer timeout", () => {
     const spy = sinon.spy(socket.reconnectTimer, "scheduleTimeout")
 
-    socket.onConnClose()
+    const event = new Event("onClose", { code: 1000 })
```

Comment:
this test is wrong, it is sending 1000 code that should not hit "scheduleTimeout", but the assert passes

Reply:
I would suggest:

```javascript
  it('does not schedules reconnectTimer timeout if normal close', () => {
    const spy = sinon.spy(socket.reconnectTimer, 'scheduleTimeout')

    const event = { code: 1000 }

    socket.onConnClose(event)

    assert.ok(spy.notCalled)
  })

  it('schedules reconnectTimer timeout if not normal close', () => {
    const spy = sinon.spy(socket.reconnectTimer, 'scheduleTimeout')

    const event = { code: 1001 }

    socket.onConnClose(event)

    assert.ok(spy.calledOnce)
  })
```